test/tools/ossfuzz/protoToYul.h

```diff
@@ -295,12 +309,16 @@ class ProtoConverter
 	}
 
 	std::ostringstream m_output;
-	/// Variables in current scope
-	std::stack<std::vector<std::string>> m_scopeVars;
+	/// Variables in all function definitions
+	std::vector<std::vector<std::vector<std::string>>> m_funcVars;
+	/// Variables in current function definition
+	std::vector<std::string const*> m_currentFuncVars;
+	/// Variables in global scope
+	std::vector<std::string const*> m_currentGlobalVars;
```

Comment:
What's the difference between this and `m_globalVars`?

Reply:
In this example

```{ let x { let y { let z, a }}}```

m_globalVars looks like so {{x}, {y}, {z,a}} where as m_currentGlobalVars (depending on where it is evaluated) is a 1d vector of live global vars at the time of evaluation e.g., within the inner most scope it is {x, y, z, a}.

The evaluation happens here: https://github.com/ethereum/solidity/pull/7729/files#diff-2cad53a043fb603d1700f10bd46f2d81R104